code/WorkInProgress/Electronics.dm

```diff
@@ -466,92 +462,245 @@
+		newsignal.data["address_1"] = "TRANSRKIT"
+		newsignal.data["sender"] = src.net_id
+		radio_connection.post_signal(src, newsignal)
 
-	//link = null
+/obj/machinery/rkit/proc/upload_blueprint(var/datum/electronics/scanned_item/O, var/target, var/internal)
+	SPAWN_DBG(0.5 SECONDS) //This proc sends responses so there must be a delay
+		var/datum/computer/file/electronics_scan/scanFile = new
+		scanFile.scannedName = O.name
+		scanFile.scannedPath = O.item_type
+		scanFile.scannedMats = O.mats
+		var/datum/signal/newsignal = get_free_signal()
+		newsignal.source = src
+		newsignal.transmission_method = TRANSMISSION_RADIO
+		if(!internal)
+			newsignal.data["command"] = "NEW"
+		else
+			newsignal.data["command"] = "UPLOAD"
+		newsignal.data["address_1"] = target
+		newsignal.data["sender"] = src.net_id
+		newsignal.data_file = scanFile
+		radio_connection.post_signal(src, newsignal)
 
-	..()
+/obj/machinery/rkit/proc/pda_message(var/target, var/message)
+	SPAWN_DBG(0.5 SECONDS) //response proc
+		var/datum/signal/newsignal = get_free_signal()
+		newsignal.source = src
+		newsignal.transmission_method = TRANSMISSION_RADIO
+		newsignal.data["command"] = "text_message"
+		newsignal.data["sender_name"] = "RKIT-MAILBOT"
+		newsignal.data["message"] = message
+		if (target) newsignal.data["address_1"] = target
+		newsignal.data["group"] = list(MGO_MECHANIC, MGA_RKIT)
+		newsignal.data["sender"] = src.net_id
+		pda.post_signal(src, newsignal)
+
+/obj/machinery/rkit/proc/transfer_database(target)
+	//If we have a database of items, and we're the host, and we see a new ruck
+	//Upload our database to it
+	var/datum/computer/file/electronics_bundle/rkitFile = new
+	rkitFile.ruckData = ruck_controls
+	rkitFile.target = target
+	SPAWN_DBG(0.5 SECONDS)
+		var/datum/signal/newsignal = get_free_signal()
+		newsignal.source = src
+		newsignal.transmission_method = TRANSMISSION_RADIO
+		newsignal.data["command"] = "UPLOAD"
+		newsignal.data["address_1"] = target
+		newsignal.data["sender"] = src.net_id
+		newsignal.data_file = rkitFile
+		radio_connection.post_signal(src, newsignal)
+	known_rucks |= target
+	send_sync()
+
+//Run this if there's a file and return
+//This will either work, or you rejected a signal that had a file it didn't need
+/obj/machinery/rkit/proc/process_upload(datum/signal/signal)
+	var/target = signal.data["sender"]
+	var/command = signal.data["command"]
+	if(!target || (command != "add" && command != "UPLOAD") || (!istype(signal.data_file, /datum/computer/file/electronics_scan) && !istype(signal.data_file, /datum/computer/file/electronics_bundle)))
+		return
+
+	var/datum/computer/file/electronics_bundle/rkitFile = signal.data_file
+	if (istype(rkitFile) && !data_initialized && !isnull(boot_time) && rkitFile.target == src.net_id) //Copy the database on digest so we never waste the effort
+		var/datum/mechanic_controller/originalData = rkitFile.ruckData
+		data_initialized = 1
+		if(world.time - boot_time <= 3 SECONDS)
```

Comment:
whatsthis

Reply:
rkitfile is a ruck database, this runs 

- if we're initializing (less than three seconds has passed since the first SYNC)
- if we haven't digested a database yet
- if our net_id was encoded into the database file by the sending ruck

I'll label it more